mapping/src/communityRelations.r

```diff
@@ -0,0 +1,475 @@
+#getDistanceByMunis <- function(muni1, muni2, map) {
+
+#cacheDistances <- function(
+
+read_mat_csv <- function(option) {
+    # Reads a csv file and returns a matrix as a data frame.
+    #
+    # Args:
+    #   option: Use '1' to read joint voting matrix.
+    #           Use '2' to read commuting matrix.
+    #
+    # Returns:
+    #   The matrix as a data frame.
+    if (option == 1)
+	mat <- read.csv("jvp_mat.csv", row.names = 1)
+    else if (option == 2)
+	mat <- read.csv("comm_mat.csv", row.names = 1)
+    else
+	stop("Must specify an option.\n1: joint voting patterns\n
+	     2: communication patterns")
+    
+    return(mat)
+}
+
+make_paired_list_from_matrix <- function(mat) {
+    # Converts matrix elements to a list of pairs.
+    #
+    # Args:
+    #   mat: The matrix to be converted.
+    # 
+    # Returns:
+    #   The paired list of values.
+    
+    # make the paired list
+    mat_as_paired_list <- as.matrix(mat) # make sure not dframe
+
+    # mat row/col pairs to row/row pairs
+    mat_as_paired_list <- melt(mat_as_paired_list)
+    
+    return(mat_as_paired_list)
+}
+
+append_comm_vals_to_jvp_list <- function(jvp_list, comm_list) {
+    # Appends commuting data column to joint voting pattern list.
+    #
+    # Args:
+    #   jvp_list: The joint voting pattern list.
+    #   comm_list: The commuting list.
+    # 
+    # Returns:
+    #   The merged list containing both jvp and comm data.
+
+    jvp_list <- cbind(jvp_list, comm_list[3])
+    return(jvp_list)
+}
+
+plot_reg_line <- function(jvp_list) {
+    # Generates a scatter plot with a linear regression line.
+    #
+    # Args:
+    #   jvp_list: The list containing muni pairs and their
+    #             corresponding jvp and comm data
+    #
+    # Returns:
+    #   Returns NULL. 
+    
+    comm <- jvp_list[, 4] # comm vec
+    jvp <- jvp_list[, 3] # voting vec
+
+    plot(comm, jvp)
+    abline(lm(jvp~comm), col = "red")  # regression line (y~x)
+    return(NULL)
+}
+
+# NOTE TO SELF: THIS FUNCTION IS NOT IN USE. DELETE??
```

Comment:
You can write `NOTE` instead of note to self [NO CHANGE REQUESTED]

Reply:
Fixed